### DISCUSSION TITLE
how to stretch object for rect,triangle, etc?

### DISCUSSION THREAD

**Discussion by KiDeokKim** [fabricjs/fabric.js#7993] [Q&A] [answered]
hi, i'm read about http://fabricjs.com/custom-controls-polygon
so i tried to stretch edge point for rect, triangle, some object like adjust polygon edge point.
it can adjust edge point using polygonPositionHandler, anchorWrapper function. but when i try to line or some object,
it doesn't work when object have not point value. ( doesn't work object list is line, some uploaded svg files )

i want adjust edge function like adobe illustrator when i selected object, and press key v and key a, switch adjust mode.

can i change rect, triangle, etc object to like polygon object?

**Comment by ShaMan123** [comment, marked as answer]
I am not sure I understand, don't use adobe illustrator.
1. Use polygon instead of rect, triangle etc.
2. try looking into `fabric.util.getPathSegmentsInfo(yourPath);`

**Reply by KiDeokKim** [reply]
okay so i solved your first reply (**Use polygon instead of rect, triangle etc.**)
but is it possible to change edge points for uploaded svg image?

**Reply by ShaMan123** [reply]
Yes, try no. 2. After you parse an svg access the path object's path property and pass it to `getPathSegmentsInfo`. And try to work with that. Pop the last piece of info as it is some kind of sum of the rest

**Reply by KiDeokKim** [reply]
i tried to read path value from **loadSVGFromString** and **groupSVGElements**. but when it have not path value, how to solve it?

here is my svg file string data : 
`<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 21.62 44.75"><defs><style>.cls-1{fill:#d3edfb;stroke-width:1.2px;}.cls-1,.cls-2,.cls-3{stroke:#000;}.cls-2,.cls-3{fill:none;}.cls-3{stroke-width:0.6px;}</style></defs><g id="layer 2" data-name="layer 2"><g id="layer_1-2" data-name="layer 1"><rect class="cls-1" x="0.61" y="7.52" width="20.41" height="28.29"/><line class="cls-2" x1="10.04" y1="35.57" x2="10.04" y2="44.75"/><polyline class="cls-3" points="9.99 7.79 9.99 12.85 14.57 12.85 14.57 16.08 6.35 16.08 6.35 19.3 14.45 19.3 14.45 22.59 6.35 22.59 6.35 25.76 14.51 25.76 14.54 29.2 9.99 29.23 10 35.57"/><line class="cls-2" x1="9.99" x2="9.99" y2="7.79"/></g></g></svg>`

**Reply by KiDeokKim** [reply]
and here is my code : load svg file
```
          fabric.loadSVGFromString(f.target.result, (objects, options) => {
            const newObjs = objects.map(el => {
              if (el.type === 'text') {
                const opts = el.toObject()
                const itext = new fabric.IText(el.text, opts)
                return itext
              }
              return el
            })
            let obj = fabric.util.groupSVGElements(newObjs, options);
            }
            obj.set({ left: canvas.getWidth() / 2, top: canvas.getHeight() / 2 }).setCoords();
            canvas.add(obj);
            canvas.renderAll();
            canvas.fire("object:modified");
          });
```

**Reply by ShaMan123** [reply]
It must have. You need to find the object in the tree if you are grouping

**Reply by ShaMan123** [reply]
load it here a see for yourself
http://fabricjs.com/kitchensink


![image](https://user-images.githubusercontent.com/34343793/172789611-2677aa99-9d8d-4859-a194-5f46cf104e86.png)

**Reply by KiDeokKim** [reply]
I apologize for troubling you.
there have 4 object in 1 group.(rect , line, polyline, line).
so i tried read path property in 4 object, but i can't find path property.

here is my image

![property](https://user-images.githubusercontent.com/40253618/172810443-c44862c8-b674-41e7-a7b2-0ad6b14c28c9.PNG)

so i change question. how to read 'path' property in canvas.getActiveObject() or fabric.util.groupSVGElements() ???

**Reply by ShaMan123** [reply]
You need to find it. It depends on your logic and UX.
If you need to support nested selection (seems so) => #7670
